cellpack/autopack/Environment.py

```diff
@@ -1700,7 +1700,24 @@ def getPointToDrop(
                 # get the most probable point using the gradient
                 # use the gradient weighted map and get mot probabl point
                 self.log.info("pick point from gradients %d", (len(allIngrPts)))
-                ptInd = self.gradients[ingr.gradient].pickPoint(allIngrPts)
+                if isinstance(ingr.gradient, list) and len(ingr.gradient) > 1:
+                    if not hasattr(ingr, "combined_weight"):
+                        gradient_list = [
+                            gradient
+                            for gradient_name, gradient in self.gradients.items()
+                            if gradient_name in ingr.gradient
+                        ]
+                        combined_weight = Gradient.get_combined_gradient_weight(
```

Comment:
can you push the logic above into a generic Gradient.pick_point_from_weight that figures out wether it needs to combine weights and returns the ptInd

Reply:
Good call! I added a static method `Gradient.pick_point_for_ingredient` [here](https://github.com/mesoscope/cellpack/pull/270/commits/6cb42589eeda1fd88b6bd5b90d6a8039bdd99181)